components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,188 @@
+package softwareversionservice
+
+import (
+	"errors"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/getosutils"
+	"github.com/chef/automate/lib/logger"
+)
+
+type ISoftwareVersionService interface {
+	GetSoftwareVersionServices(string) (models.SoftwareVersionDetails, error)
+}
+
+type SoftwareVersionService struct {
+	cmdCheckArray []string
+	osFilepath    string
+	logger        logger.Logger
+}
+
+func NewSoftwareVersionService(logger logger.Logger) ISoftwareVersionService {
+	return &SoftwareVersionService{
+		cmdCheckArray: cmdCheckArray,
+		osFilepath:    osFilepath,
+		logger:        logger,
+	}
+}
+
+const (
+	availability = " availability"
+	Ensure       = "Ensure "
+	osFilepath   = "/etc/os-release"
+)
+
+var cmdCheckArray = []string{"mkdir", "useradd", "chown", "rm", "touch", "truncate", "echo", "sleep", "ls", "grep", "yum", "which", "cp", "curl", "bash", "sysctl", "cat", "sed", "mount", "pvcreate", "vgcreate", "lvcreate", "mv", "systemd", "wget", "exec"}
+
+func (sv *SoftwareVersionService) GetSoftwareVersionServices(query string) (models.SoftwareVersionDetails, error) {
```

Comment:
Please use pointers to struct while receiving or returning from function

Reply:
Done